### ISSUE TITLE
Document how to setup for home-manager's NixOS module

### ISSUE THREAD

**Issue by ametis70** [nix-community/nixd#626]
**Is your feature request related to a problem? Please describe.**
I read the documentation and I understood how to setup completion for NixOS and home-manager options when Home-Manager is installed in it's standalone mode, but I cannot make the options show up when using the NixOS Module

**Describe the solution you'd like**
I'd like to make the options to show up. Is it even possible? If so, I believe it should be documented alongside the other examples.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

Thanks in advance!

**Comment by inclyc** [comment]
> I'd like to make the options to show up. Is it even possible?

Yes, it is possible. You can refer to https://github.com/nix-community/nixd/issues/615#issue-2647914685 for a simple solution, (i.e. use `getSuboptions`).

>  If so, I believe it should be documented alongside the other examples.

Yes, we can add it to the documentation, or if you figured it out, feel free to make PRs.

**Comment by ametis70** [comment]
Great, that worked perfectly. I'll make a PR to add it to the documentation. Thanks!

**Comment by wvffle** [comment]
For some reason it doesn't work for me. I get this error whenever nixd starts evaluates through LSP:


```
home-manager eval expr: -32001: The option `<name>.home.stateVersion' was accessed but has no value defined. Try setting the option.
```

I have the `home.stateVersion` set in my home-manager configuration.


Here's how I tried to configure nixd:
```nix
          options = rec {
            nixos.expr = "${flake}.nixosConfigurations.${hostname}.options";
            home-manager.expr = "${nixos.expr}.home-manager.users.type.getSubOptions []";
          };

```

**Comment by wvffle** [comment]
@louisdutton that's exactly what I did to workaround the issue. However I'd rather it would be fixed than adding one more configuration just for the sake of nixd

**Comment by Noah765** [comment]
I've tracked down this problem and found that in my case, stylix along with how I set `home.stateVersion` was causing it.

You can read more about the specifics of the problem [here](https://github.com/danth/stylix/issues/1098), but the gist is that stylix tries to access `home.stateVersion` while declaring an option, which fails if `home.stateVersion` isn't defined in a module defined by `home-manager.sharedModules`.

A simple workaround for this is either not to use stylix, or to define `home.stateVersion` inside of a shared home manager module like this:
```nix
home-manager.sharedModules = [{home.stateVersion = "<value>";}];
```

I am sure that there are other similar causes of this problem.

**Comment by Brisingr05** [comment]
Until recently, I was also using the `homeConfigurations` output *just* for HM completions as mentioned above[^1], because following the instructions added in https://github.com/nix-community/nixd/pull/686 didn't show all HM options. Specifically, completions for my own custom modules as well as modules provided by third-party projects (e.g. `homeModules.catppuccin` from https://github.com/catppuccin/nix) were not showing using the `getSubOptions` method. Turns out, the solution was to move those modules from the `imports` section in my Home Manager config to `home-manager.sharedModules`. Now there's no need for creating a `homeConfigurations` flake output.

[^1]: https://github.com/nix-community/nixd/issues/626#issuecomment-2764258421
